### PR TITLE
Committing tests, changed ALERT_SEVERITY and ALERT_TYPE to enums

### PR DESCRIPTION
**Pull request by samir-elhelw76**
Adding my unit tests, note it references a directory "test_data" so this directory should be created locally if you want to run my tests.

### CODE REVIEW

File: tests/signature_tests/test_ipsignature.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from scapy.all import rdpcap
+from scapy.layers.inet import IP
+from src.signature_detection.ip_signature import IPSignature
+from pathlib import Path
+
+
+class TestIPSignature(unittest.TestCase):
+    def setUp(self) -> None:
+        self.ip_signature = IPSignature("192.168.1.0/24")
+        packets = rdpcap(str(Path("../tests/test_data/test_tcpdump2.pcap").resolve()))
```
Comment (briweinstein) [inline tests/signature_tests/test_ipsignature.py:11]: Is this pcap too big to add in the PR?

File: tests/signature_tests/test_ipsignature.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        self.src_malicious_ip_packet = packets[0]
+        self.dst_malicious_ip_packet = packets[1]
+        self.src_benign_ip_packet = packets[2]
+        self.dst_benign_ip_packet = packets[3]
+        self.src_malicious_ip_packet[IP].src = "62.210.205.141"
```
Comment (briweinstein) [inline tests/signature_tests/test_ipsignature.py:16]: What are these IPs?

File: tests/signature_tests/test_ipsignature.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    def test_benign_ip(self):
+        self.assertFalse(self.ip_signature(self.src_benign_ip_packet))
+        self.assertFalse(self.ip_signature(self.dst_benign_ip_packet))
+
+    def test_malicious_ip(self):
+        self.assertTrue(self.ip_signature(self.src_malicious_ip_packet))
+        self.assertTrue(self.ip_signature(self.dst_malicious_ip_packet))
+
+    def tearDown(self) -> None:
```
Comment (briweinstein) [inline tests/signature_tests/test_ipsignature.py:29]: ?

File: tests/signature_tests/test_ipsignature.py (hunk 1 above)
Comment (samir-elhelw76) [reply tests/signature_tests/test_ipsignature.py:11]: It's not in the actual merge, to run these tests you can create a pcap with this name locally

File: tests/signature_tests/test_ipsignature.py (hunk 2 above)
Comment (samir-elhelw76) [reply tests/signature_tests/test_ipsignature.py:16]: Known malicious IPs that hit on virustotal

File: tests/signature_tests/test_ipsignature.py (hunk 3 above)
Comment (samir-elhelw76) [reply tests/signature_tests/test_ipsignature.py:29]: I don't understand what's being asked in this comment

File: tests/test_main_loop.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    def setUp(self) -> None:
+        packets = rdpcap(str(Path("test_data/test_tcpdump2.pcap").resolve()))
+        self.src_malicious_ip_packet = packets[0]
+        self.dst_malicious_ip_packet = packets[1]
+        self.src_malicious_ip_packet[IP].src = "62.210.205.141"
+        self.dst_malicious_ip_packet[IP].dst = "5.188.62.140"
+        self.ids_signatures = [IPSignature("192.168.1.0/24"), MACAddressSignature()]
+        self.signature_detector = SignatureDetector(self.ids_signatures)
+
+    def test_pull_and_validate_addrs(self):
```
Comment (briweinstein) [inline tests/test_main_loop.py:26]: Why is this here?

Comment (samir-elhelw76) [reply tests/test_main_loop.py:26]: It was a template, was going to implement actual tests later for test_pull_and_validate_addrs

File: tests/test_main_loop.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import unittest
```
Comment (briweinstein) [inline tests/test_main_loop.py:1]: You're not calling the other test files from here, where will they be called?

File: tests/signature_tests/test_ipsignature.py (hunk 1 above)
Comment (briweinstein) [reply tests/signature_tests/test_ipsignature.py:11]: But the pcap has to be a specific file so the first two are malicious and the second two are benign. If it's small enough can you add it so all the tests are working off the same file?

File: tests/signature_tests/test_ipsignature.py (hunk 3 above)
Comment (briweinstein) [reply tests/signature_tests/test_ipsignature.py:29]: What is tearDown? I'm not familiar with the python function

File: tests/test_main_loop.py (hunk 4 above)
Comment (briweinstein) [reply tests/test_main_loop.py:26]: Pulling from the config file? Or pulling from where?

Comment (briweinstein) [reply tests/test_main_loop.py:26]: Can you add a TODO so we don't forget it's not finished?

File: tests/test_main_loop.py (hunk 5 above)
Comment (samir-elhelw76) [reply tests/test_main_loop.py:1]: The tests can be ran from the python CLI or within an IDE, I don't have a file that calls all of them in this commit though.

**Review by MartinLPetrauskas — APPROVED** [review]
Tested this PR. OpenWrt package compiles successfully and does not crash when run.

File: src/dashboard/alerts/alert.py [hunk 6]
```diff
@@ -7,15 +7,16 @@
 from scapy.utils import hexdump
 from src.emailalerts import emailsystem
 from src import run_config as CONFIG
+from enum import Enum, IntEnum
 
 
-class ALERT_TYPE:
+class ALERT_TYPE(Enum):
```
Comment (lucas-m98) [inline src/dashboard/alerts/alert.py:13]: Didn't even realize I was doing this wrong. Thanks 👍

**Comment by lucas-m98** [comment]
Tests look good. Only issue is the .pcap file. We could share it offline or in the repo, I'm fine with either but it'll probably be necessary to have it locally at some point.

**Review by lucas-m98 — APPROVED** [review]

**Review by briweinstein — APPROVED** [review]

File: setup/tinyHIPPOstart [hunk 7]
```diff
@@ -14,21 +14,21 @@ pip3 install scapy
...
-export IOT_IDS_CONFIGPATH='/etc/tinyHIPPO/config.json'
+# set install path environment variable
+export TINYHIPPO_INSTALL_PATH='/etc/tinyHIPPO/'
 
 # Move the dashboard files
 # TODO: this is a hack to get tinyHIPPO running for the midterm progress report
-cp -r /etc/tinyHIPPO/src/dashboard/luasrc/controller/ids-priv /usr/lib/lua/luci/controller
-cp -r /etc/tinyHIPPO/src/dashboard/luasrc/view/ids-priv /usr/lib/lua/luci/view
+cp -r $TINYHIPPO_INSTALL_PATH/src/dashboard/luasrc/controller/ids-priv /usr/lib/lua/luci/controller
+cp -r $TINYHIPPO_INSTALL_PATH/dashboard/luasrc/view/ids-priv /usr/lib/lua/luci/view
```
Comment (MartinLPetrauskas) [inline setup/tinyHIPPOstart:23]: This copy command throws an error. Should be $TINYHIPP_INSTALL_PATH/src/dashboard/.....

File: config.py [hunk 8]
```diff
@@ -14,17 +16,20 @@ class ConfigEmail:
 
 class Config:
     def __init__(self):
+        self.install_path = Path(os.getenv("TINYHIPPO_INSTALL_PATH"))
```
Comment (MartinLPetrauskas) [inline config.py:19]: How does Path work specifically? And the exporting of environment variables. It seems like the environment variable only exists when tinyHIPPO is running, and otherwise is not. Maybe I need to do more research into how environment variables work, but some clarification would be appreciated

**Review by MartinLPetrauskas — CHANGES_REQUESTED** [review]
Change is needed for the proper file path of the one of the lua src directories

File: config.py (hunk 8 above)
Comment (samir-elhelw76) [reply config.py:19]: The enviornment variable should last for the duration of the programs runtime, and since this variable is set in the start script it should also be reset in the case of a reboot. Path is a library that provides an OS dependent way of navigating file systems by creating a filepath object that can be queried for various properties. It comes built in to python3 so it doesn't create any additional overhead.

File: setup/tinyHIPPOstart [hunk 9]
```diff
@@ -14,21 +14,21 @@ pip3 install scapy
 pip3 install dataclasses
 pip3 install requests
 
-# set config path environment variable
-export IOT_IDS_CONFIGPATH='/etc/tinyHIPPO/config.json'
+# set install path environment variable
+export TINYHIPPO_INSTALL_PATH='/etc/tinyHIPPO/'
```
Comment (MartinLPetrauskas) [inline setup/tinyHIPPOstart:18]: When I attempted to install and run, the output of 'ps' showed this:
```
10549 root     21120 R    python3 /etc/tinyHIPPO//tiny_hippo_run.py
```
I don't think there needs to be two forward slashes, so change it to only one.

Comment (samir-elhelw76) [reply setup/tinyHIPPOstart]: Pushed these changes
